rfd/0059-search-based-access-requests.md

```diff
@@ -0,0 +1,278 @@
+---
+authors: Alexander Klizhentas (sasha@goteleport.com), Nic Klaassen (nic@goteleport.com)
+state: draft
+---
+
+# RFD 59 - Search Based Access Requests
+
+## What
+
+This RFD proposes just-in-time search-based access requests which will enable
+users to request access to resources based on a search and/or selection of
+individual resources, rather than having to request one or more roles.
+
+## Why
+
+Teleport users may not know in advance which roles they need or who should grant
+access, just which resource(s) they need access to.
+
+Users would like to avoid having many different custom roles for different
+levels of access.
+
+Requesting access to a "blanket" role violates the principle of least priviledge.
+Most of the time incident responders only need access to 1 or 2 nodes,
+search-based access requests will allow them to request access to only the nodes
+they need.
+
+
+## Details
+
+Users will be able to request access by using search, and roles will be used
+behind the scenes to evaluate who will be responsible for reviewing and granting
+access requests.
+
+Users won’t be asked to think about roles in advance, but will be able to focus
+on finding and requesting access to individual resources.
+
+The existing “role-based” access requests and the new “search-based” access
+requests proposed here are useful for two different scenarios:
+
+- Sometimes it is helpful to request elevated roles in the system, like
+  `db-admin`, to perform some system upgrades, or system wide-troubleshooting.
+  In this case, role-based access requests work great.
+
+- In other cases, it’s helpful to request access to a subset of resources. In
+  this case, search.
+
+### Example Flow
+
+Alice is a member of the group “splunk”.
+
+She gets an alert that “db-1” is malfunctioning.
+
+Alice needs to locate the "db-1" resource and request access to it.
+
+Alice goes to the “Request access” screen and selects: “select access to resources”
+option that is available alongside the “select access to roles” option.
+
+Alice uses standard (fuzzy search) or advanced search (labels) to enter the
+resource name.
+
+She finds a single resource or a group of resources by that name and clicks
+"request access".
+
+Teleport identifies that access to the node “db-1” is managed by team
+“db-admins” based on Alice's role which allowed her to search for this node.
```

Comment:
I'm confused, how does Alice's role that allowed her to search for this node enable teleport to identify that access to it is managed by "db-admins"?

Reply:
this sentence was wrong, the access request will be sent to the right channel based on the resource labels, we'll be updating the plugins to support this, clarified in the doc